connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoIngestionUtils.scala

```diff
@@ -0,0 +1,69 @@
+package com.microsoft.kusto.spark.utils
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility
+import com.fasterxml.jackson.annotation.PropertyAccessor
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.microsoft.azure.kusto.ingest.IngestionMapping.IngestionMappingKind
+import com.microsoft.azure.kusto.ingest.{ColumnMapping, IngestionProperties}
+import com.microsoft.azure.kusto.ingest.IngestionProperties.DATA_FORMAT
+import com.microsoft.kusto.spark.datasink.SchemaAdjustmentMode
+import com.microsoft.kusto.spark.datasink.SchemaAdjustmentMode.SchemaAdjustmentMode
+import org.apache.spark.sql.types.StructType
+import org.json.JSONObject
+
+object KustoIngestionUtils {
+
+  private[kusto] def adjustSchema(mode: SchemaAdjustmentMode ,sourceSchema: StructType, targetSchema: Array[JSONObject], ingestionProperties: IngestionProperties) : Unit = {
+
+    mode match {
+      case SchemaAdjustmentMode.NoAdjustment =>
+      case SchemaAdjustmentMode.FailIfNotMatch => forceAdjustSchema(sourceSchema, targetSchema)
+      case SchemaAdjustmentMode.GenerateDynamicCsvMapping => setCsvMapping(sourceSchema, targetSchema, ingestionProperties)
+    }
+
+  }
+
+  private[kusto] def forceAdjustSchema(sourceSchema: StructType, targetSchema: Array[JSONObject]) : Unit ={
+
+    val targetSchemaColumns = targetSchema.map(c=> c.getString("Name")).toSeq
+    val sourceSchemaColumns = sourceSchema.map(c=> c.name)
+
+    if (targetSchemaColumns != sourceSchemaColumns) {
+      throw new RuntimeException(s"Target table schema does not match to DataFrame schema. " +
```

Comment:
We are moving from using RuntimeException - you can see RetriesExhaustedException for example - lets make a new exceptions folder while we are at it and here call all these exceptions SchemaMatchException or something similar